### DISCUSSION TITLE
Limitation

### DISCUSSION THREAD

**Discussion by rudnypc** [KiraLT/torrent-stream-server#241] [Q&A]
Hi,

When I get some torrents, I saw no peers found, I have to wait some minutes before I get more torrents.

**Comment by KiraLT** [comment]
Hello,

What version do you use? `v1.9.0` has a bug fix related to `no peers found` errors.  It's related to upload limitations, so you can also try removing the upload limit by setting ENV variable (or adding it to [config](https://github.com/KiraLT/torrent-stream-server/wiki/Configuration)):

```python
TORRENTS_UPLOAD_LIMIT=-1
```

Also if there are only a few peers it is possible that they don't reply fast and you trigger timeout.

**Reply by rudnypc** [reply]
How much magnet links per second the system support  to work?

**Reply by rudnypc** [reply]
after 3 request at the same time, I get always no peers found, I have to restart the container.

**Reply by KiraLT** [reply]
Technically there is no limit (it depends on server hardware) unless the issue is related to the upload limit bug. I will need to do some performance tests and I will come back to this question.

**Reply by KiraLT** [reply]
I opened an issue regarding the speed limitation issue - #244. In theory, it shouldn't affect if you adding a few torrents/second. But there will be limitations when adding more - so you should disable upload limitations in that case. I also noticed some strange pattern when I got `no peers found` error, but app reload solved this issue. I will need to do more investigations.
